test/index.js

```diff
@@ -1,257 +1,415 @@
-      const args = ['install', 'chromedriver=invalid'];
-      console.log = mockConsoleLog;
-      console.error = mockConsoleError;
-      await browserDriverManager(args);
+describe('browser-driver-manager', () => {
+  describe('which', async () => {
+    it('logs the locations of chrome and chromedriver if they exist', async () => {
+      await makeEnvFile();
+      await which();
 
-      expect(mockConsoleError.calledWith(sinon.match.string, 'Chrome version "invalid" is not a number')).to.be.true;
+      sinon.assert.calledWith(mockConsoleLog, envContents);
     });
-
-    it('should correctly get major version of chrome', async () => {
-      const args = ['install', 'chromedriver'];
-      mockStdoutChunk = 'Google Chrome 97.0.4692.71';
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(mockConsoleLog.calledWith('Installing ChromeDriver 97')).to.be.true;
+    it('errors if no environment file exists', async () => {
+      await expect(which()).to.be.rejectedWith(
+        'No environment file exists. Please install first'
+      );
     });
+  });
+  describe('version', () => {
+    it('logs the version when a valid one exists', async () => {
+      await makeEnvFile();
+      await version();
 
-    it('should fetch npm chromedriver version', async () => {
-      const args = ['install', 'chromedriver', '--verbose'];
-      mockStdoutChunk = 'Google Chrome 97.0.4692.71';
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(chromdriverNock.isDone()).to.be.true;
+      sinon.assert.calledWith(mockConsoleLog, mockVersion);
     });
-
-    it('should use the version if it is equal to the latest npm chromedriver', async () => {
-      const args = ['install', 'chromedriver=97'];
-      const expected = ['install', '--no-save', 'chromedriver@97'];
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(spawnStub.calledWith('npm', expected)).to.be.true;
+    it('errors if no environment file exists', async () => {
+      await expect(which()).to.be.rejectedWith(
+        'No environment file exists. Please install first'
+      );
     });
-
-    it('should use the version if it is below to the latest npm chromedriver', async () => {
-      const args = ['install', 'chromedriver=96'];
-      const expected = ['install', '--no-save', 'chromedriver@96'];
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(spawnStub.calledWith('npm', expected)).to.be.true;
+  });
+  describe('install', () => {
+    describe('creates', () => {
+      it("the cache directory if it doesn't already exist", async () => {
+        await install(browser);
+        await expect(fsPromises.access(MOCK_BDM_CACHE_DIR)).to.be.fulfilled;
+      });
+
+      describe('the environment file when', () => {
+        it('a valid version is given', async () => {
```

Comment:
This and the next test aren't really testing the code so much as they are testing your mocks. Since Puppeteer is responsible for giving an error if the version or channel passed is invalid, any test mocking that behavior will not reflect how the code will behave in actuality. This will be the case for all Puppeteer apis that are mocked as well (for example trying to say that it'll throw an error for invalid versions).

Probably the better way to handle this is just to ensure the mock installs were both called when calling with a valid browser. You won't be able to test different channels or versions since Puppeteer handles resolving all those so you'll need to instead write tests that described / test the functionality that resolves around what Puppeteer returns.